messages_appendix.cddl

```diff
@@ -549,7 +505,159 @@ receive-video-capability = {
   ? 8: bool ; supports-scaling
 }
 
-receive-data-capability = {
-  0: format ; format
+; type key 134
+streaming-session-start-request = {
+  request
+  1: uint; streaming-session-id
+  2: [* media-stream-offer] ;  stream-offers
+  3: microseconds ; stats-interval
+}
+
+; type key 135
+streaming-session-start-response = {
+  response
+  1: &result ; result
+  2: [* media-stream-request] ; stream-requests
+  3: microseconds ; stats-interval
+}
+
+; type key 136
+streaming-session-modify-request = {
+  request
+  1: uint ; streaming-session-id
+  2: [* media-stream-request] ; stream-requests
+}
+
+; type key 137
+streaming-session-modify-response = {
+  response
+  1: &result ; result
+}
+
+; type key 138
+streaming-session-terminate-request = {
+  request
+  1 : uint ; streaming-session-id
+}
+
+; type key 139
+streaming-session-terminate-response = {
+  response
+}
+
+; type key 140
+streaming-session-terminate-event = {
+  1 : uint ; streaming-session-id
+}
+
+media-stream-offer = {
+  1 : uint ; media-stream-id
+  ? 2 : text ; friendly-name
+  ? 3 : [1* audio-encoding-offer] ; audio
+  ? 4 : [1* video-encoding-offer] ; video
+}
+
+media-stream-request = {
+  1 : uint ; media-stream-id
+  ? 2 : audio-encoding-request ; audio
+  ? 3 : video-encoding-request ; video
+}
+
+audio-encoding-offer = {
+  1 : uint ; encoding-id
+  2 : text ; codec
+  3 : uint ; time-scale
+  ? 4 : uint ; default-duration
+}
+
+video-encoding-offer = {
+  1 : uint ; encoding-id
+  2 : text ; codec
+  3 : uint ; time-scale
+  ? 4 : uint ; default-duration
+  ? 5 : video-rotation ; default-rotation
+}
+
+audio-encoding-request = {
+  1: uint ; encoding-id
+}
+
+video-encoding-request = {
+  1 : uint ; encoding-id
+  ? 2 : video-resolution ; target-resolution
+  ? 3 : float32 ; max-frames-per-second
+}
+
+; type key 13
+audio-frame = [
+  encoding-id: uint
+  start-time: uint
+  payload: bytes
+  ? optional: {
+    ? 0: uint ; duration
+    ? 1: media-time ; sync-time
+  }
+]
+
+; type key 14
+video-frame = {
+  0: uint ; encoding-id
+  1: uint ; sequence-number
+  ? 2: [* int] ; depends-on
+  3: uint ; start-time
```

Comment:
The spec says that start-time and duration for an offer are inferred from the properties of the offer.  But it doesn't say exactly how.

Reply:
Time-scale and duration may be if if omitted.  But not start-time, and not if omitted.

Reply:
Sorry I didn't quite follow that response, maybe restate it?

Reply:
Time-scale and duration may be inferred.  

start-time cannot be inferred.

Inference is basically just "if it isn't there, use the default", but I guess I could make that more explicit.

Reply:
Made more explicit now.